cli/cobra.go

```diff
@@ -234,9 +234,13 @@ func isExperimental(cmd *cobra.Command) bool {
 }
 
 func additionalHelp(cmd *cobra.Command) string {
-	if additionalHelp, ok := cmd.Annotations["additionalHelp"]; ok {
+	if msg, ok := cmd.Annotations["additionalHelp"]; ok {
+		out := cmd.OutOrStderr()
+		if _, isTerminal := term.GetFdInfo(out); !isTerminal {
```

Comment:
Had this as a branch in my local repo for some time, but had to dust it off a bit (previously depended on https://github.com/docker/cli/pull/3971, but I rewrote it to detect the terminal slightly different)